components/dm/spec/logic.go

```diff
@@ -326,37 +327,36 @@ func (i *DMMasterInstance) InitConfig(e executor.Executor, clusterName, clusterV
 }
 
 // ScaleConfig deploy temporary config on scaling
-func (i *DMMasterInstance) ScaleConfig(e executor.Executor, b *spec.Specification, clusterName, clusterVersion, deployUser string, paths meta.DirPaths) error {
-	panic("TODO")
-	/*
-		if err := i.instance.InitConfig(e, clusterName, clusterVersion, deployUser, paths); err != nil {
-			return err
-		}
+func (i *DMMasterInstance) ScaleConfig(e executor.Executor, topo spec.Topology, clusterName, clusterVersion, deployUser string, paths meta.DirPaths) error {
+	if err := i.InitConfig(e, clusterName, clusterVersion, deployUser, paths); err != nil {
+		return err
+	}
 
-		c := b
-		spec := i.InstanceSpec.(MasterSpec)
-		cfg := scripts.NewDMMasterScaleScript(
-			spec.Name,
-			i.GetHost(),
-			paths.Deploy,
-			paths.Data[0],
-			paths.Log,
-		).WithPort(spec.Port).WithNumaNode(spec.NumaNode).WithPeerPort(spec.PeerPort).AppendEndpoints(c.Endpoints(deployUser)...)
-
-		fp := filepath.Join(paths.Cache, fmt.Sprintf("run_dm-master_%s_%d.sh", i.GetHost(), i.GetPort()))
-		log.Infof("script path: %s", fp)
-		if err := cfg.ConfigToFile(fp); err != nil {
-			return err
-		}
+	c := topo.(*DMTopologySpecification)
```

Comment:
Can we simplify the name `DMTopologySpecification`? Or just `Specification` or `Topology` is enough? The two specification definitions are placed in a different package, I think it's ok to use the same type name.